src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -437,81 +420,21 @@ final case class ClientApiServiceImpl(
-
-      if (condition) Future.successful(true)
-      else Future.failed(SecurityOperatorRelationshipNotActive(relationshipId))
-    }
-
-    for {
-      relationship <- partyManagementService.getRelationshipById(relationshipId)
-      _            <- isActiveSecurityOperatorRelationship(relationship)
-    } yield relationship
-  }
-
-  private[this] def securityOperatorRelationship(
-    requesterId: UUID,
+  private[this] def getSecurityUser(
+    selfcareId: UUID,
+    requesterUserId: UUID,
     userId: UUID,
-    roles: Seq[String] =
-      Seq(PartyManagementService.PRODUCT_ROLE_SECURITY_OPERATOR, PartyManagementService.PRODUCT_ROLE_ADMIN)
-  )(implicit contexts: Seq[(String, String)]): Future[PartyManagementDependency.Relationship] = for {
-    selfcareId <- tenantManagementService.getTenantById(requesterId).flatMap(_.selfcareId.toFuture(MissingSelfcareId))
-    relationships <- partyManagementService
-      .getRelationships(selfcareId, userId, roles)
-    activeRelationShips = relationships.items.toList.filter(
-      _.state == PartyManagementDependency.RelationshipState.ACTIVE
-    )
-    relationShip <- activeRelationShips.headOption.toFuture(SecurityOperatorRelationshipNotFound(requesterId, userId))
-  } yield relationShip
-
-  private[this] def operatorsFromClient(client: PersistentClient)(implicit
-    contexts: Seq[(String, String)]
-  ): Future[Seq[Operator]] = Future.traverse(client.relationships.toList)(operatorFromRelationship)
-
-  private[this] def operatorFromRelationship(
-    relationshipId: UUID
-  )(implicit contexts: Seq[(String, String)]): Future[Operator] =
-    for {
-      relationship <- partyManagementService.getRelationshipById(relationshipId)
-      user         <- userRegistryManagementService.getUserById(relationship.from)
-      userInfo     <- extractUserInfo(user)
-      (name, familyName, taxCode) = userInfo
-      operatorState <- relationshipStateToApi(relationship.state).toFuture
-    } yield Operator(
-      relationshipId = relationship.id,
-      taxCode = taxCode,
-      name = name,
-      familyName = familyName,
-      role = relationshipRoleToApi(relationship.role),
-      product = relationshipProductToApi(relationship.product),
-      state = operatorState
-    )
-
-  def extractUserInfo(user: UserResource): Future[(String, String, String)] = {
-    val userInfo = for {
-      name       <- user.name
-      familyName <- user.familyName
-      fiscalCode <- user.fiscalCode
-    } yield (name.value, familyName.value, fiscalCode)
-
-    userInfo.toFuture(MissingUserInfo(user.id))
-  }
+    roles: Seq[String] = Seq(SECURITY_ROLE, ADMIN_ROLE)
+  )(implicit contexts: Seq[(String, String)]): Future[Unit] = for {
+    users <- selfcareV2ClientService
+      .getInstitutionProductUsers(selfcareId, requesterUserId, userId.some, roles)
+      .map(_.map(_.toApi))
+    _     <- users.headOption.toFuture(SecurityUserNotFound(requesterUserId, userId))
```

Comment:
It is not required to convert to api model.
If you want to double-check that the returned value is valid, we can flatMap on the id, like
```suggestion
    users <- selfcareV2ClientService
      .getInstitutionProductUsers(selfcareId, requesterUserId, userId.some, roles)
      .map(_.flatMap(_.id))
    _     <- users.headOption.toFuture(SecurityUserNotFound(requesterUserId, userId))
```

This means you can also delete [this](https://github.com/pagopa/interop-be-authorization-process/pull/232/files#diff-934273d235108ff79b9f25c8155c28ea0398d4802cb7efd3bd132f6cf50609deR263)